### DISCUSSION TITLE
Recursive Whisper fails at python

### DISCUSSION THREAD

**Discussion by boludoz** [openai/whisper#613] [Q&A]
Recursive Whisper fails at python mode, I am using Whisper on a RTX 3080 12gb and it fails to decode the second file, there is no way to free the memory of the video card, the same happens in google colab, I think it is a memory leak.

**Comment by arabcoders** [comment]
I am using the python module, and doing hundreds of files, you are only supposed to load the module once and keep on calling transcribe method on already loaded module, you dont need to reload the module on each file. for example 

```python
import whisper
model = whisper.load_model('large')

files_list = []

for file in files_list:
    result = model.transcribe(file)
    print(result["text"])
```

**Reply by dgoryeo** [reply]
Can this "loop" approach be done in Google Colab notebook? 
I don't have powerfull enough GPU in my local PC, so I need to run Whisper on colab notebook.

**Reply by arabcoders** [reply]
> Can this "loop" approach be done in Google Colab notebook? 
> 
> I don't have powerfull enough GPU in my local PC, so I need to run Whisper on colab notebook.
> 
> 

I assume it can, hut not sure google colab works.

**Reply by dgoryeo** [reply]
Somewhere I had read that Whisper model needs to be reset for each input file. I'm not sure why. Doing the reset in Colab is not that straightforward.
